apps/hash-frontend/src/lib/user-and-org.ts

```diff
@@ -148,6 +148,12 @@ export const constructOrg = (params: {
 }): Org => {
   const { subgraph, orgEntity } = params;
 
+  if (orgEntity.metadata.entityTypeId !== types.entityType.org.entityTypeId) {
+    throw new Error(
+      `Entity with type ${orgEntity.metadata.entityTypeId} is not an org entity`,
+    );
+  }
+
```

Comment:
How this is possible? If this is only possible via a bug it would be good to state that in a comment or remove it again (I guess this is how you found the bug?).

Reply:
This was possible because we cast from `Entity` to `Entity<UserProperties>` (or `Entity<OrgProperties`) when we pass the entities returned by the structural queries to these functions. In this case it was possible for this bug to occur because the `useOrgWithLinks` and `useUserWithLinks` hooks were querying for entities via their `UUID` only, and did not include checks for the type. I've added these checks in https://github.com/hashintel/hash/pull/3444/commits/1613bf37eee44fad54cab7b1caaeb65daa160371 to make these hooks more robust, but still think having an error here could help catch when an incorrect cast has been made more easily.

Reply:
Great, thank you!